### PR TITLE
Enables space ruin generation in the station z-level. Adds new ruins for the sake of station ruins.

### PR DESCRIPTION
**Pull request by BlackMajor**
This PR basically does what it says on the tin.

-Enables space ruins to generate in the station z-level.
-Ruins permitted to generate are in their own folder with respective config.
-Headmins will need to update configs accordingly, since PR's can't edit those.
-Quick note. These asteroids WILL nerf the danger of meteors simply by existing to take some of the blows.

Why make this?
Honestly I've always wanted the space around the station not to feel so empty. It's nice to have reference points to work with in station space. As well as a reason for miners to mine in SPACE.

However I must maintain that all ruins in the station z are just flavoursome things or asteroids. No gamer gear, no dangerous things. Just peculiarities. Maybe a toolbox or insuls or two, but nothing major.

Ruins currently prepared are as follows. (Ore scanners not included)

roid1.dmm. It's just a rock. Repeatable.
image

roid2.dmm. A rock, but with crabs (who are raving).
image

roid3.dmm. A ruin with a little farm house and potential space dorm.
Image of Roid3

roid4.dmm. A collection of rocks. Featuring laz injector.
image

roid5.dmm. A rock with a hollowed out middle and some remains. The middle has air in it. Poor soul.
image

roid6.dmm. A crashed escape pod. Contains a spear and spider meat.
image

roid7.dmm. A derelict bar in an asteroid.
image

roid8.dmm. Contains a coconut bong, wiz robes and a toilet with an arcane wand in it. A wand that fires 20 burn projectiles. Fairly weak, should be ok?
image

roid9.dmm. A roid with a crew monitoring console, security camera console, and a dorm. A good place to hide from the station.
image

roid10.dmm. A little maze with a drill and dogborg plush.
image

More to come, give me ideas.

### CODE REVIEW

**Comment by Owai-Seek** [comment]
Maybe an abandoned bar + kitchen with scattered junkfood wrappers, blood, and broken glasses. Shotgun shells scattered around and the corpse of a dead bartender in the back (kitchen) with an empty medkit, its contents tossed across the floor.

A somewhat larger escape pod with a dead dead spider or two and some dead crew on it. (Think the pubby pod. or bigger.) Maybe a bear trap or improve weapons, and a dead person in a broken soft spacesuit.

**Comment by Ghommie** [comment]
Outside making the meteors events less relevant (which I honestly hardly care since it's almost a waste of time now), This may also impact shuttles (mostly dynamic ones) as well a few other things.

**Comment by BlackMajor** [comment]
> Outside making the meteors events less relevant (which I honestly hardly care since it's almost a waste of time now), This may also impact shuttles (mostly dynamic ones) as well a few other things.

In initial testing, the static shuttles just flex on the ruins and land ontop of them if anything crosses their landing zone, as expected.

As for the dynamic shuttles, yeah. You can't set them to land where the roids spawn. Ideally there won't be so many roids that you can't land a shuttle where you want them to be. They're not dense enough at the moment to cloud possible landing sites. But that's what testing's for.

Still gonna need more ruins to spawn, though.

**Comment by Ghommie** [comment]
Also from the screenshots it seems you have mapped a lot of rock walls with high chance of ores. It seems pretty excessive considering it's definitely safer than the mining zone.

**Comment by BlackMajor** [comment]
> Also from the screenshots it seems you have mapped a lot of rock walls with high chance of ores. It seems pretty excessive considering it's definitely safer than the mining zone.

Fair. My thought process is that miners don't get space suits at round start.
Ideally I want it to be viable to go mine asteroids. I can bump the rarity down though so it's not as rewarding.

File: code/modules/mapping/ruins.dm [hunk 1]
```diff
@@ -6,7 +6,11 @@
 		sanity--
 		var/width_border = TRANSITIONEDGE + SPACERUIN_MAP_EDGE_PAD + round(width / 2)
 		var/height_border = TRANSITIONEDGE + SPACERUIN_MAP_EDGE_PAD + round(height / 2)
-		var/turf/central_turf = forced_turf ? forced_turf : locate(rand(width_border, world.maxx - width_border), rand(height_border, world.maxy - height_border), z)
+		var/turf/central_turf
+		if(is_station_level(z)) //used to make a square around the station for generating station side ruins without getting too close.
+			central_turf = pick(block(locate(10,10,z), locate((world.maxx - 10), (world.maxy - 10), z)) - block(locate(45,45,z), locate(210,210,z))) //FIGURE OUT A BETTER WAY TO DO THIS
```
Comment (Ghommie) [inline code/modules/mapping/ruins.dm]: Just make a `/area/space/noruins` with `NO_RUINS_1` in their `flags_1` field mapped all around the station surrounding tiles.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/mapping/ruins.dm (hunk 1 above)
Comment (BlackMajor) [reply code/modules/mapping/ruins.dm]: Ghom out here with the 400IQ solution I was too smol brained to think of.

**Comment by Ghommie** [comment]
Please include the new area on other station maps where possible (Omegastation included) and reduce the ore availability from the asteroid rocks if you haven't done so already.
Let's not fuel the awful community's disinterest toward other maps.

Also more merge conflicts.

**Comment by BlackMajor** [comment]
> Please include the new area on other station maps where possible (Omegastation included) and reduce the ore availability from the asteroid rocks if you haven't done so already.
> Let's not fuel the awful community's disinterest toward other maps.
> 
> Also more merge conflicts.

Yeah I'm gonna.

**Comment by Ghommie** [comment]
More conflicts, and KiloStation.

Also, you could have used the mapmerger script to avoid all these line changes.

**Comment by MimicFaux** [comment]
I like the idea of bits of space junk and old satellites drifting through the station's z-level. Perhaps some fancier (and rarer) things like a ruskie radio brick with a dismantled telecomms system on it. I also like the ruined bar. Bits of old/other stations that didn't make it.

**Review by Ghommie — APPROVED** [review]

**Comment by Ghommie** [comment]
> _maps/RandomRuins/SpaceRuinsStation/roid9.dmm:22:error: bad type or variable: DormRoid1

Fix it pretty please.

**Comment by Ghommie** [comment]
Oh well, even if you used the mapmerger, the amount of lines changed would still be obscenely high.

**Pull request merged**